src/utils/helpers/list/buildQueryFilters.ts

```diff
@@ -78,10 +121,23 @@ const constructFilter = (
   if (substituteColumnName) return { [substituteColumnName]: filter }
 
   // otherwise use filterName
-  return { [filterName]: filter }
+  const gqlFilterName = options?.searchFields ? options.searchFields[0] : filterName
+  return { [gqlFilterName]: filter }
 }
 
 const splitCommaList = (values: string) => values.split(',')
 
 // Use this if the values can be free text strings (e.g. stage name)
-const inList = (values: string) => ({ inInsensitive: splitCommaList(values) })
+const inList = (values: string, searchField: string, nullString: string) => {
+  const valuesArray = splitCommaList(values)
+  if (!valuesArray.includes(nullString) || searchField === '')
+    return { inInsensitive: replaceCommasArray(valuesArray) }
+
+  const nonBlankValues = valuesArray.filter((val) => val !== nullString)
+  return {
+    or: [
+      { [searchField]: { inInsensitive: replaceCommasArray(nonBlankValues) } },
+      { [searchField]: { isNull: true } },
+    ],
+  }
+}
```

Comment:
Yep. Looks like it was fun indeed. But this part if very difficult to read that's for sure. I wonder it would be worth just adding those new fields `searchField` and `nullString` as optionals and set a default value. Wouldn't it be simple?

Reply:
I would have, but you can only have one optional argument. :( 

I would kind of like to re-write some more of this, but I was very conscious not to break existing ApplicationList functionality.

Reply:
Ah -- there it is, now I can see the comment again. Odd 🤔

Anyway, changed the input parameters to an object, which is much better for handling multiple option arguments.
b322d6e8